app/views/sessions/purchase_cart.html.erb

```diff
@@ -7,7 +7,11 @@
     </tr>
     <% if session[:cart] %>
         <% session[:cart].each do |item_id, quantity| %>
-        <% item = Item.find_by(id: item_id) %>
+            <% item = Item.find_by(id: item_id) %>
+            <!-- Line below generates receipt for purchasing this item, if possible -->
+            <% if logged_in? %>
```

Comment:
First, you want the URL code for the receipt to resemble a UUID, so 1) it's both hard for other people to guess, and 2) it masks the true number of receipts / transactions that the site has hosted (i.e., you don't want to reveal site statistics).

Second, this wouldn't be clunky. Even on a phone, the URL could just be emailed to them. There wouldn't be any need for copy/paste.